datatransfer/impl/graphsync/graphsync_impl_test.go

```diff
@@ -281,13 +281,9 @@ func TestDataTransferValidation(t *testing.T) {
 	require.NoError(t, err)
```

Comment:
I would read out the buffer bytes here and have createDTRequest take a byte array called selectorBytes. Otherwise it's unclear what's supposed to be in the buffer.